pkg/cmd/psa/psa.go

```diff
@@ -0,0 +1,302 @@
+	o.warnings = &warningsHandler{}
+	restConfig.WarningHandler = o.warnings
+	o.client, err = kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create client: %w", err)
+	}
+
+	return nil
+}
+
+// Run attempts to update the namespace psa enforce label to the psa audit value.
+func (o *PSAOptions) Run() error {
+	// Get a list of all the namespaces.
+	namespaceList, err := o.getNamespaces()
+	if err != nil {
+		return fmt.Errorf("failed to get namespaces: %w", err)
+	}
+
+	podSecurityViolations := []*PodSecurityViolation{}
+	// Gather all the warnings for each namespace, when enforcing audit-level.
+	for _, namespace := range namespaceList.Items {
+		psv, err := o.checkNamespacePodSecurity(&namespace)
+		if err != nil {
+			return fmt.Errorf("failed to check namespace %q: %w", namespace.Name, err)
+		}
+		if psv == nil {
+			continue
+		}
+
+		klog.V(4).Infof(
+			"Pod %q has pod security violations, gathering Pod and Deployment Resources",
+			psv.PodName,
+			namespace.Name,
+		)
+
+		psv.Pod, err = o.client.CoreV1().
+			Pods(namespace.Name).
+			Get(context.Background(), psv.PodName, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf(
+				"failed to get pod %q from %q, which violates %q: %w",
+				psv.PodName,
+				namespace.Name,
+				psv.Violations[0],
+				err,
+			)
+		}
+
+		psv.PodController, err = o.getPodController(psv.Pod)
+		if err != nil {
+			return fmt.Errorf("failed to get pod controller: %w", err)
+		}
+
+		podSecurityViolations = append(podSecurityViolations, psv)
+	}
+
+	if len(podSecurityViolations) == 0 {
+		return nil
+	}
+
+	// Print the violations.
+	if err := json.NewEncoder(os.Stdout).Encode(podSecurityViolations); err != nil {
+		return err
+	}
+
+	if !o.quite {
+		os.Exit(1)
+	}
+
+	return nil
+}
+
+// checkNamespacePodSecurity collects the pod security violations for a given
+// namespace on a stricter pod security enforcement.
+func (o *PSAOptions) checkNamespacePodSecurity(ns *corev1.Namespace) (*PodSecurityViolation, error) {
+	nsCopy := ns.DeepCopy()
+
+	// Update the pod security enforcement for the dry run.
+	nsCopy.Labels[psapi.EnforceLevelLabel] = o.level
+
+	klog.V(4).Infof("Checking nsCopy %q for violations at level %q", nsCopy.Name, o.level)
+
+	// Make a server-dry-run update on the nsCopy with the audit-level value.
+	_, err := o.client.CoreV1().
+		Namespaces().
+		Update(
+			context.Background(),
+			nsCopy,
+			metav1.UpdateOptions{DryRun: []string{"All"}},
+		)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the warnings from the server-dry-run update.
+	warnings := o.warnings.PopAll()
+	if len(warnings) == 0 {
+		return nil, nil
+	}
+
+	return parseWarnings(warnings), nil
+}
+
+// getNamespaces returns the namespace that should be checked for pod security.
+// It could be given by the flag. Defaults to all namespaces.
+func (o *PSAOptions) getNamespaces() (*corev1.NamespaceList, error) {
+	if o.namespace == "" {
```

Comment:
standard behavior is to honor the kubeconfig namespace.

Reply:
Yes, should work.